independent-projects/arc/runtime/src/main/java/io/quarkus/arc/AbstractInvocationContext.java

```diff
@@ -52,32 +46,36 @@ protected AbstractInvocationContext(Object target, Method method, Function<Invoc
         return interceptorBindings;
     }
 
-    @Override
-    public abstract Object proceed() throws Exception;
-
-    protected static Map<String, Object> newContextData(Set<Annotation> interceptorBindings) {
-        Map<String, Object> result = new HashMap<String, Object>();
-        result.put(InvocationContextImpl.KEY_INTERCEPTOR_BINDINGS, interceptorBindings);
-        return result;
-    }
-
     @Override
     public Method getMethod() {
         return method;
     }
 
     @Override
     public Object[] getParameters() {
-        if (parameters == null) {
-            throw new IllegalStateException();
-        }
         return parameters;
     }
 
     @Override
-    public void setParameters(Object[] params) throws IllegalStateException, IllegalArgumentException {
-        if (parameters == null) {
-            throw new IllegalStateException();
+    public void setParameters(Object[] params) {
+        int newParametersCount = Objects.requireNonNull(params).length;
+        Class<?>[] parameterTypes = method.getParameterTypes();
+        if (parameterTypes.length != newParametersCount) {
+            throw new IllegalArgumentException(
+                    "Wrong number of parameters - method has " + Arrays.toString(parameterTypes) + ", attempting to set "
```

Comment:
I think it might help to have some quotes around the parameter types (both of them) ?

Reply:
`Arrays.toString()` is using `[]` so I don't think it's necessary.